### PR TITLE
implement ConversionToAssetBalance in asset-rate

### PR DESCRIPTION
**Pull request by girazoki**
Implements the `ConversionToAssetBalance` trait to asset-rate by doing `1/rate*(balance)`.

### CODE REVIEW

File: substrate/frame/asset-rate/src/lib.rs
Comment (joepetrowski) [inline substrate/frame/asset-rate/src/lib.rs:250]:
```suggestion
/// Exposes conversion of a native balance to an asset balance.
```

**Review by joepetrowski — APPROVED** [review]

**Review by muharem — APPROVED** [review]

**Comment by muharem** [comment]
@girazoki can you please provide a prdoc - https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/prdoc.md

**Comment by girazoki** [comment]
Do not merge yet as I have to take into account the case in which we set a 0 conversion and therefore there is an overflow

File: prdoc/pr_2903.prdoc
Comment (gpestana) [inline prdoc/pr_2903.prdoc:4]:
```suggestion
title: "Implement `ConversionToAssetBalance` in asset-rate"
```

File: prdoc/pr_2903.prdoc
Comment (gpestana) [inline prdoc/pr_2903.prdoc:11]:
```suggestion
      Previously only the `ConversionFromAssetBalance` trait was implemented, which would allow to convert an asset balance into the corresponding native balance.
```

File: substrate/frame/asset-rate/src/lib.rs
Comment (gpestana) [inline substrate/frame/asset-rate/src/lib.rs:153]:
```suggestion
		/// Overflow ocurred when calculating the inverse rate.
```

File: substrate/frame/asset-rate/src/lib.rs
Comment (gpestana) [inline substrate/frame/asset-rate/src/lib.rs:272]:
```suggestion
		// We cannot use `saturating_div` here so we use `checked_div`.
```

**Comment by girazoki** [comment]
Is there any issue finally merging this?

**Comment by girazoki** [comment]
@gpestana I think we nede one more review

**Comment by girazoki** [comment]
@bkchr can we get one more review on this PR please?

**Comment by bkchr** [comment]
You have all the required reviews. Just no one pressed the merge button.

**Comment by girazoki** [comment]
> You have all the required reviews. Just no one pressed the merge button.

Ah sorry, I though I needed one more! Thanks!

**Comment by girazoki** [comment]
> You have all the required reviews. Just no one pressed the merge button.

I dont think I broke the westend runtime upgrade with this PR no?

**Pull request merged**
